cpp/src/io/avro/avro.cpp

```diff
@@ -30,7 +30,7 @@ namespace avro {
  *
  * @returns true if successful, false if error
  */
-bool container::parse(file_metadata *md, size_t max_num_rows, size_t first_row)
+bool container::parse(file_metadata &md, size_t max_num_rows, size_t first_row)
```

Comment:
The cudf convention is to take `out` params by pointer so as to indicate at call site that this will be modified.

Reply:
Sounds good. I just checked to see if we can turn this in to something along the lines of `std::pair<bool, file_metadata> parse(int, int);`, as the invocation site suggests it can, but it requires combining the reader impl constructor+read functions to make it pretty. Saving for another time, and reverting to `file_metadata *md`